src/main/java/com/myhome/services/springdatajpa/UserSDJpaService.java

```diff
@@ -39,33 +40,36 @@
  */
 @Service
 @Slf4j
+@RequiredArgsConstructor
 public class UserSDJpaService implements UserService {
 
   private final UserRepository userRepository;
   private final UserMapper userMapper;
   private final PasswordEncoder passwordEncoder;
-
-  @Autowired
-  private CommunityService communityService;
-
-  public UserSDJpaService(UserRepository userRepository,
-      UserMapper userMapper,
-      PasswordEncoder passwordEncoder) {
-    this.userRepository = userRepository;
-    this.userMapper = userMapper;
-    this.passwordEncoder = passwordEncoder;
-  }
+  private final CommunityService communityService;
 
   @Override public UserDto createUser(UserDto request) {
     generateUniqueUserId(request);
     encryptUserPassword(request);
     return createUserInRepository(request);
   }
 
-  @Override public Set<User> listAll() {
+  @Override
+  public Set<User> listAll() {
+    return listAll(null, null);
```

Comment:
Don't we need a default value for these params? At least for limit? Let me know your thoughts.

Reply:
im not familiar too much with the logic of this app so I can't be sure if we need it, i don't see any reason to limit the fetch.
however it is possible to use Pageable objects and limit results as seen [here ](https://stackoverflow.com/questions/33018308/limit-findall-possible-with-spring), i can transform the Set to Pageable. do you think it is needed?

Reply:
Sounds like a good solution to use pages. We will do it in a separate request. Thanks!